pkg/kubelet/api/v1alpha1/runtime/api.proto

```diff
@@ -0,0 +1,629 @@
+    required string runtime_api_version = 4;
+}
+
+message StatusRequest {
+}
+
+message StatusResponse {
+}
+
+// PodSandboxConfig holds all the required and optional fields for creating a
+// sandbox.
+message PodSandboxConfig {
+    // Name is the name of the sandbox.
+    required string name = 1;
+    // Hostname is the hostname of the sandbox.
+    required string hostname = 2;
+    // Path to the directory on the host in which container log files are
+    // stored.
+    // By default the Log of a container going into the LogDirectory will be
+    // hooked up to STDOUT and STDERR. However, the LogDirectory may contain
+    // binary log files with structured logging data from the individual
+    // containers. For example the files might be newline seperated JSON
+    // structured logs, systemd-journald journal files, gRPC trace files, etc.
+    // E.g.,
+    //     PodSandboxConfig.LogDirectory = `/var/log/pods/<podUID>/`
+    //     ContainerConfig.LogPath = `containerName_Instance#.log`
+    //
+    // WARNING: Log managment and how kubelet should interface with the
+    // container logs are under active discussion in
+    // https://issues.k8s.io/24677. There *may* be future change of direction
+    // for logging as the discussion carries on.
+    required string log_directory = 3;
+    // DNSOptions sets the DNS options for the sandbox.
+    optional DNSOption dns_options = 4;
+    // PortMappings lists the port mappings for the sandbox.
+    repeated PortMapping port_mappings = 5;
+    // Resources specifies the resource limits for the sandbox (i.e., the
+    // aggregate cpu/memory resources limits of all containers).
+    // Note: On a Linux host, kubelet will create a pod-level cgroup and pass
+    // it as the cgroup parent for the PodSandbox. For some runtimes, this is
+    // sufficent. For others, e.g., hypervisor-based runtimes, explicit
+    // resource limits for the sandbox are needed at creation time.
+    optional PodSandboxResources resources  = 6;
+    // Labels are key value pairs that may be used to scope and select individual resources.
+    map<string, string> labels = 7;
+    // Annotations is an unstructured key value map that may be set by external
+    // tools to store and retrieve arbitrary metadata.
+    map<string, string> annotations = 8;
+    // Linux contains configurations specific to Linux hosts.
+    optional LinuxPodSandboxConfig linux = 9;
+}
+
+// DNSOption specifies the DNS servers and search domains.
+message DNSOption {
+    // Servers is a list of DNS servers of the cluster.
+    repeated string servers = 1;
+    // Searches is a list of DNS search domains of the cluster.
+    repeated string searches = 2;
+}
+
+// PortMapping specifies the port mapping configurations of sandbox
+message PortMapping {
+    // Name of the port mapping
+    optional string name = 1;
+    // Protocol of the port mapping.
+    // Valid candidates are: TCP, UDP
+    required string protocol = 2;
```

Comment:
Good idea, thanks a lot. I will replace all those types into enums.


2016-06-16 12:00 GMT+08:00 James DeFelice <notifications@github.com>:

> In pkg/kubelet/api/v1alpha1/runtime/api.proto
> <https://github.com/kubernetes/kubernetes/pull/17048#discussion_r67284257>
> :
>
> > +
> > +// DNSOption specifies the DNS servers and search domains.
> > +message DNSOption {
> > +    // Servers is a list of DNS servers of the cluster.
> > +    repeated string servers = 1;
> > +    // Searches is a list of DNS search domains of the cluster.
> > +    repeated string searches = 2;
> > +}
> > +
> > +// PortMapping specifies the port mapping configurations of sandbox
> > +message PortMapping {
> > +    // Name of the port mapping
> > +    optional string name = 1;
> > +    // Protocol of the port mapping.
> > +    // Valid candidates are: TCP, UDP
> > +    required string protocol = 2;
>
> enum's are pretty explicit, perhaps protocol could be based on such?
> … <#m_2928675505780144151_>
> On Wed, Jun 15, 2016 at 8:23 PM, Pengfei Ni ***@***.***> wrote: In
> pkg/kubelet/api/v1alpha1/runtime/api.proto <#17048 (comment)
> <https://github.com/kubernetes/kubernetes/pull/17048#discussion_r67269582>>
> : > + > +// DNSOption specifies the DNS servers and search domains. >
> +message DNSOption { > + // Servers is a list of DNS servers of the
> cluster. > + repeated string servers = 1; > + // Searches is a list of DNS
> search domains of the cluster. > + repeated string searches = 2; > +} > + >
> +// PortMapping specifies the port mapping configurations of sandbox >
> +message PortMapping { > + // Name of the port mapping > + optional string
> name = 1; > + // Protocol of the port mapping. > + // Valid candidates are:
> TCP, UDP > + required string protocol = 2; Do you know how to do so in
> protobuf? There is no syntax to define const strings in protobuf. — You are
> receiving this because you were mentioned. Reply to this email directly,
> view it on GitHub <
> https://github.com/kubernetes/kubernetes/pull/17048/files/094e05a2fb10072606b6e212ff4a25955eaacffd#r67269582>,
> or mute the thread <
> https://github.com/notifications/unsubscribe/ACPVLPdDNrZqH7R_NJrNmzefdZKd1HUhks5qMJd_gaJpZM4GfQ_B>
> .
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/17048/files/094e05a2fb10072606b6e212ff4a25955eaacffd#r67284257>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AApTHSdvt_9SebSCnJI7nAwR8zgYPEzBks5qMMpRgaJpZM4GfQ_B>
> .
>